service.go

```diff
@@ -102,29 +99,27 @@ func (s *Service) listSubnets(ctx context.Context) ([]net.IPNet, error) {
 	return existingSubnets, nil
 }
 
-// NewSubnet returns the next available subnet, of the configured size,
+// CreateSubnet returns the next available subnet, of the configured size,
 // from the configured network.
-func (s *Service) NewSubnet(mask net.IPMask) (net.IPNet, error) {
-	s.logger.Log("info", "creating new subnet")
+func (s *Service) CreateSubnet(ctx context.Context, mask net.IPMask, annotation string) (net.IPNet, error) {
```

Comment:
New is a name indicating a constructor. This function has side effects. I renamed to Create. It's api breaking change so not a big deal.